packages/trie/src/trie/trie.ts

```diff
@@ -670,8 +677,8 @@ export class Trie {
    * Creates a proof from a trie and key that can be verified using {@link Trie.verifyProof}.
    * @param key
    */
-  async createProof(key: Buffer): Promise<Proof> {
-    const { stack } = await this.findPath(this.appliedKey(key))
+  async createProof(key: Buffer, ignoreAppliedKey = false): Promise<Proof> {
```

Comment:
I'm debating to remove this property and to just add API for:

`createProof(key: Buffer)`

And

`private _createProof(key: Buffer, ignoreAppliedKey = false)`

Reply:
Note that without this API we get proof for the wrong keys on hashed tries, if we try to proof key X then it is actually going to prove `hash(key)` which has value `null`. (We also do not have the preimages)